File: FormulaBollo/src/shared/models/sprint.ts
import { Driver } from "./driver";
import { Race } from "./race";
import { SprintPosition } from "./sprintPosition";

export class Sprint {
  race: Race;
  driver: Driver;
  position: SprintPosition | null;

  constructor(
    race: Race,
    driver: Driver,
    position: SprintPosition | null
  ) {
    this.race = race;
    this.driver = driver;
    this.position = position;
  }
}

File: FormulaBolloApp/src/main/java/formula/bollo/entity/Sprint.java
package formula.bollo.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "sprint")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Sprint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, length = 7)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "race_id")
    private Race race;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "driver_id")
    private Driver driver;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "position_id")
    private SprintPosition position;
}